tests/compile-test.rs

```diff
@@ -444,7 +444,12 @@ impl DiagnosticCollector {
                     iter::zip(DEPRECATED, DEPRECATED_VERSION)
                         .map(|((lint, reason), version)| LintMetadata::new_deprecated(lint, reason, version)),
                 )
+                .map(|mut metadata| {
+                    metadata.docs = sanitize_explanation(&metadata.docs);
+                    metadata
+                })
```

Comment:
Good catch that this should apply to config too, currently https://rust-lang.github.io/rust-clippy/master/index.html#/explicit_iter_loop doesn't get syntax highlighted because of this

Ideally it would be done in `get_configuration_metadata` though so that `--explain` also gets the cleaned up version of config docs

Reply:
Done, I had to move `sanitize_explanation` to `clippy_config`, but the configuration docs are accounted for!